user_service/src/controllers/UserController.ts

```diff
@@ -43,4 +44,54 @@ export const delUserProfile = async (req: any, res: any) => {
             error: error,
             message: 'User not found.'})
     }
+}
+
+
+//@desc     get user's answered questions
+//@route    GET /api/users/:id/questions
+//@access   authenticated users
+export async function getUserQuestions(req: any, res: any) {
+  try {
+    const userId = req.params.id
+
+    if (isNaN(userId)) {
+      return res.status(400).json({ error: 'Invalid user ID' });
+    }
+
+    const answeredQuestions = await prisma.answeredQuestion.findMany({
+      where: {
+        userId: userId,
+      },
+    });
+
+    return res.json(answeredQuestions);
+  } catch (error) {
+    console.error(error);
+    return res.status(500).json({ error: 'Internal Server Error' });
+  }
+}
+
+
+
+//@desc     add a new user question
+//@route    POST /api/users/:id/addquestions
+//@access   authenticated users
+export async function addUserQuestion(req: any, res: any) {
+  try {
+    const { userId, questionId, complexity, category } = req.body;
+
+    const createdQuestion = await prisma.answeredQuestion.create({
+      data: {
+        userId: userId,
+        questionId: questionId,
+        complexity: complexity,
+        category: { set: category },
+      },
+    });
```

Comment:
On closer inspection, this does not come anywhere close to adding a user's answeredQuestion to the database at all

The schema describes the following fields, but complexity is specified instead of difficulty. Also, the `solved` field is not supplied as well.

```
  id         Int        @id @default(autoincrement())
  userId     Int
  user       User     @relation(fields: [userId], references: [id])
  questionTitle String
  questionId Int
  difficulty Int
  category   String[]
  solved     Boolean
  answeredAt DateTime @default(now())
  ```